modules/library/main/src/main/java/org/geotools/feature/DefaultFeatureCollection.java

```diff
@@ -182,7 +182,7 @@ protected boolean add(SimpleFeature feature, boolean fire) {
         }
         SimpleFeatureType childType = (SimpleFeatureType) getSchema();
         if (!feature.getFeatureType().equals(childType)) {
-            LOGGER.warning("Feature Collection contains a heterogeneous" + " mix of features");
+            LOGGER.fine("Feature Collection contains a heterogeneous mix of features");
```

Comment:
This test is behaving as intended, and should probably be upgraded to an exception if the warning is not being sufficient

Long term we should change this to an exception and arrange an helper class for FeatureCollection that supports mixed content as I agree we will see more of these over time.

Is there a helper class in app schema that we can make use of right now?

Reply:
I was thinking if we could make the class have two "modes", one for when AbstractFeatureType is used and then relax this check as @aaime indicated. 

Sadly the class explicitly asks getSchema() to return SimpleFeatureType so we cannot return AbstractFeatureType here